### PR TITLE
Feat/provider - the one with hooks

### PR DESCRIPTION
**Pull request by jgfidelis**
Please test this on: 
https://fidelis--delivery.myvtex.com/

Ja está sem o OrderFormProvider na StoreWrapper
#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [ ] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- Use a lot more hooks.
+- Get rid of OrderForm context.
```
Comment (klzns) [inline CHANGELOG.md]: Add in another section `### Removed`

File: react/components/AddressList.js [hunk 2]
```diff
@@ -1,29 +1,47 @@
...
+  }, [address])
+
+  if (availableAddresses.length === 0) {
+    return null
+  }
+
+  return (
+    <div className="vtex-address-locator__address-list pa5">
+      <FormattedMessage id="address-manager.address-list">
+        {text => <span className="t-small f6 dark-gray">{text}</span>}
```
Comment (klzns) [inline react/components/AddressList.js]: `dark-gray` is deprecated use `c-muted-1` instead

File: react/components/AddressListItem.js [hunk 3]
```diff
@@ -29,20 +19,25 @@ export default class AddressListItem extends Component {
...
-        <p className="ma0 t-body c-on-base">{`${street}, ${number}`}</p>
-        <span className="f7 t-small c-muted-2">{!neighborhood ? city : `${neighborhood}, ${city}` }</span>
-      </div>
-    )
-  }
+
+  const { street, number, neighborhood, city } = address
+  return (
+    <div
+      className={`${!isLastAddress && 'bb b--light-gray'} pv4 dim pointer`}
```
Comment (klzns) [inline react/components/AddressListItem.js]: `b--light-gray` is deprecated use `b--muted-3` or 4 instead

File: react/index.js [hunk 4]
```diff
@@ -1,74 +1,38 @@
...
-
-    /** TODO: Add a redirect placeholder (perhaps one of those "If you are not redirected, click here" things)
-     * @author lbebber */
-    return (
-      <AddressPage loading />
-    )
-  }
+  useEffect(() => {
+    checkIfAddressIsSet()
+  })
```
Comment (klzns) [inline react/index.js]: I think it might make sense to add `address` here:
```suggestion
  }, [address])
```

Comment (jgfidelis) [reply react/index.js]: it definitely does

File: react/components/AddressChallenge/index.js [hunk 5]
```diff
@@ -1,103 +1,59 @@
...
-          </div>
-        </div>
-      )
-    }
-
-    if (hasAddress) {
-      return children
+const AddressChallenge = ({ data, children }) => {
+  const { navigate, page } = useRuntime()
+  const [hasAddress, updateHasAddress] = useState(false)
```
Comment (klzns) [inline react/components/AddressChallenge/index.js]:
```tsx
const hasAddress = !loading && !!(shippingData && shippingData.address)
```

File: react/components/AddressChallenge/index.js [hunk 6]
```diff
@@ -1,103 +1,59 @@
...
+  const [isLoading, updateLoading] = useState(true)
```
Comment (klzns) [inline react/components/AddressChallenge/index.js]: Use loading instead

File: react/components/AddressManager/AddressPage.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      <div
+        className="vtex-address-modal__address-page"
+        style={transformAnimationStyle('110%', isPickupOpen && isMobile)}
+      >
+        <Card>
+          <ExtensionPoint
+            id="address-search"
+            onUpdateOrderForm={handleOrderFormUpdated}
+          />
+          {/* <AddressSearch onUpdateOrderForm={handleOrderFormUpdated} /> */}
```
Comment (klzns) [inline react/components/AddressManager/AddressPage.js]: Remove

File: react/components/AddressManager/AddressPage.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+          <ExtensionPoint
+            id="pickup-selector"
+            onConfirm={handleOrderFormUpdated}
+            onFindPickupClick={handlePickupClick}
+            isPickupOpen={isPickupOpen}
+            parentRef={isMobile && wrapperRef}
+            closeModal={closeModal}
+          />
+          {/* <PickupSelector
```
Comment (klzns) [inline react/components/AddressManager/AddressPage.js]: Remove

File: react/components/AddressOrderFormProvider.js [hunk 9]
```diff
@@ -1,36 +1,54 @@
...
+        ...pick(['orderForm', 'loading', 'refetch'], addressQuery),
         updateOrderFormProfile,
         updateOrderFormShipping,
         updateOrderFormCheckin,
-      }
+      },
     }
-  }, [orderFormId, orderFormQuery.loading])
-  return <AddressContext.Provider value={value}>{children}</AddressContext.Provider>
+  }, [addressQuery])
```
Comment (klzns) [inline react/components/AddressOrderFormProvider.js]: Missing from list of dependencies

updateOrderFormProfile,
updateOrderFormShipping,
updateOrderFormCheckin,

File: react/components/AddressRedeem/AddressRedeemQueryWrapper.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        },
+      })
+      console.log('teste response: ', response)
+      hasError = Boolean(
+        path(['data', 'updateOrderFormProfile', 'orderForm'], response)
+      )
+    } catch (e) {
+      hasError = true
+    }
+    console.log('teste haserror: ', hasError)
```
Comment (klzns) [inline react/components/AddressRedeem/AddressRedeemQueryWrapper.js]: remove

File: react/components/AddressRedeem/AddressRedeemQueryWrapper.js [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+  const updateAddressProfile = async ({ email }) => {
+    let hasError = false
+    try {
+      const response = await address.updateOrderFormProfile({
+        variables: {
+          orderFormId: address.orderForm.orderFormId,
+          fields: { email },
+        },
+      })
+      console.log('teste response: ', response)
```
Comment (klzns) [inline react/components/AddressRedeem/AddressRedeemQueryWrapper.js]: remove

File: react/components/AddressRedeem/AddressRedeemQueryWrapper.js [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+ */
+
+const AddressRedeemQueryWrapper = ({ rules, client }) => {
+  const {
+    culture: { country },
+  } = useRuntime()
+  const { address } = useAddress()
+  const [queryLoading, updateQueryLoading] = useState(false)
+  const [profile, updateProfile] = useState(
+    addValidation({ homePhone: '' }, rules)
```
Comment (klzns) [inline react/components/AddressRedeem/AddressRedeemQueryWrapper.js]: use lazy initializer

File: react/components/PickupSelector/PickupModalContainer.js [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+        updatePickupOptions([])
+        updatePickupPoints([])
+        updateSearching(false)
+        return
+      }
+      const pickupOptions = pickupPoints.map(convertPickupPointToOption)
+      const newSelectedPickupPoint = pickupOptions[0]
+      updatePickupOptions(pickupOptions)
+      updatePickupPoints(pickupPoints)
+      updateSearching(false)
```
Comment (klzns) [inline react/components/PickupSelector/PickupModalContainer.js]: When you have this situation where you find many setStates together is time to use useReducer instead.

File: react/components/ChangeAddress/index.js [hunk 14]
```diff
@@ -1,187 +1,90 @@
...
-    this.handleCloseModal()
-  }
-
-  handleChangeAddress = () => {
-    this.handleOrderFormUpdated()
-  }
-
-  handleCloseModal = () => {
-    const { closeModal } = this.props
+// margin 0 -1.5rem -3rem -1.5rem
```
Comment (lbebber) [inline react/components/ChangeAddress/index.js]: ?

**Pull request merged**